internal/configs/virtualserver.go

```diff
@@ -302,25 +322,19 @@ func generateVirtualServerConfig(virtualServerEx *VirtualServerEx, tlsPemFileNam
 	}
 }
 
-func generateUpstream(upstreamName string, upstream conf_v1alpha1.Upstream, endpoints []string, isPlus bool, cfgParams *ConfigParams) version2.Upstream {
+func generateUpstream(upstreamName string, upstream conf_v1alpha1.Upstream, externalNameSvcs map[string]bool, endpoints []string, cfgParams *ConfigParams) version2.Upstream {
 	var upsServers []version2.UpstreamServer
 
+	// isExternalNameSvc is always false for OSS
+	_, isExternalNameSvc := externalNameSvcs[upstream.Service]
```

Comment:
Looks like there is a bug in the `externalNameSvcs` map because the keys contain names without namespaces. 

Consider a (corner) case if we have 2 VirtualServerRoutes in different namespaces and they both refer to a service with the same name. If in one namespace the service is of the externalname type, the IC will think that the service in the second namespace is also of the externalname type.

Reply:
Yeah I had the namespaces in the first implementation, then I removed them cos I saw we didn't use them for externalname services for Ingress Resources. Adding them back.